packages/core/src/utils.ts

```diff
@@ -1232,23 +1333,34 @@ export const getCanonicalConfigArtifacts = async (
         compilerOutput.contracts?.[sourceName]?.[contractName]
 
       if (contractOutput !== undefined) {
-        const creationCodeWithConstructorArgs =
-          getCreationCodeWithConstructorArgs(
-            add0x(contractOutput.evm.bytecode.object),
-            contractConfig.constructorArgs,
-            referenceName,
-            contractOutput.abi
-          )
-
+        // Construct the initial artifact without the creation code or the complete bytecode
+        // Since those need to have external libraries injected which we will do next
         artifacts[referenceName] = {
-          creationCodeWithConstructorArgs,
+          creationCodeWithConstructorArgs: '',
           abi: contractOutput.abi,
           compilerInput,
           compilerOutput,
           sourceName,
           contractName,
           bytecode: add0x(contractOutput.evm.bytecode.object),
+          linkReferences: contractOutput.evm.bytecode.linkReferences,
         }
+
+        // update the bytecode with the external libraries
+        artifacts[referenceName].bytecode =
+          await getBytecodeWithExternalLibraries(
+            artifacts[referenceName],
+            contractConfig.libraries
+          )
+
+        // update creation code with linked bytecode
+        artifacts[referenceName].creationCodeWithConstructorArgs =
+          getCreationCodeWithConstructorArgs(
+            artifacts[referenceName].bytecode,
+            contractConfig.constructorArgs,
+            referenceName,
+            contractOutput.abi
+          )
```

Comment:
This section can be cleaned up if we refactor `getBytecodeWithExternalLibraries` to only accept the fields of the ContractArtifact that it actually uses, instead of passing in the entire ContractArtifact. After doing this, we can call `getBytecodeWithExternalLibraries` initially, then create the `artifacts[referenceName]` object in a single assignment instead of three assignments